### PR TITLE
Typescript initial implementation

### PR DESCRIPTION
**Pull request by ggaabe**
This implements all the packages and build settings for running TypeScript, and switches the index.js file to an index.ts file. Allows us to slowly convert all code to typescript as needed without interfering with current javascript implementations.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,22 +18,27 @@
...
     "binance-api-node": "^0.4.3",
     "cheerio": "^1.0.0-rc.2",
     "cryptocompare": "^0.3.0",
     "discord.js": "^11.2.1",
     "eightball": "^1.0.1",
     "node-coinmarketcap-api": "^1.0.4",
     "node-fetch": "^1.7.3",
+    "nodemon": "^1.17.1",
     "r2": "^2.0.0",
+    "ts-node": "^5.0.1",
```
Comment (edreesjalili) [inline package.json]: ts-node, nodemon, and @types need to be in dev deps

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+const replies = require('./replies')
+const swears = require('./swears')
+const market = require('./market')
+const trade = require('./trade')
+const ticker = require('./ticker')
+const recommend = require('./recommend')
+const predict = require('./predict')
+const config = require('./config')
+const eightball = require('eightball')
+const logger = require('./logger')
```
Comment (edreesjalili) [inline src/index.ts]: In TS, we should be doing import calls not requires

File: src/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (edreesjalili) [inline src/tsconfig.json]: remove this file, its a dup

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "lib": ["es2015", "es2015.promise", "dom"],                             /* Specify library files to be included in the compilation. */
+    "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    "sourceMap": true,                     /* Generates corresponding '.map' file. */
+    // "outFile": "./",                       /* Concatenate and emit output to single file. */
+    // "outDir": "./",                        /* Redirect output structure to the directory. */
```
Comment (edreesjalili) [inline tsconfig.json]: outDir should be dist/

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
```
Comment (edreesjalili) [inline tsconfig.json]: src/

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
+/**
+ * index.js
+ *
+ * Application entry point.
+ */
```
Comment (edreesjalili) [inline src/index.ts]: we can probably get rid of this comment

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+export interface Global {
```
Comment (edreesjalili) [inline src/index.ts]: Why do we need this?

**Comment by ggaabe** [comment]
Global doesn't work otherwise, and node-fetch was set on global.fetch in
the original .js line

File: src/index.js [hunk 8]
```diff
@@ -1,97 +1,106 @@
...
-const client = new Discord.Client()
+"use strict";
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
```
Comment (nazimamin) [inline src/index.js]: Do we need these? Since we're using index.ts revert all these to base.

**Comment by edreesjalili** [comment]
@ggaabe Remove the global stuff, just because it was done a certain way before doesn't mean we have to keep doing it that way. Just import it like the others, doesn't need to be global. Also that global interface you made, im pretty sure its for like either angular or react code.

File: src/index.js (hunk 8 above)
Comment (ggaabe) [reply src/index.js]: These are just generated, idk what it does, but it will be generated every time we run

**Comment by ggaabe** [comment]
I deleted the comments, global stuff, and made the outdir dist. This is just so we can start with Typescript. If you see anything else wrong, I think you can modify the commit directly

**Pull request merged**